scripts/configure.py

```diff
@@ -1176,6 +1176,8 @@ def failed():
     pre_backup = _create_backup(env, "_pre-fw-upgrade")
     progress(pre_backup)
     tasks += pre_backup
+  if failed():
+    return False
```

Comment:
considering we've had a spate of failed upgrades recently, and that i've changed the upgrade script most recently here with backups, this was where I started looking. It's of interest, because this pre-backup happens before we install requirements w/pip, but would not necessary fail until later - obviating the actual source of the issue.